go-controller/hybrid-overlay/pkg/util/util.go

```diff
@@ -0,0 +1,94 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+)
+
+// ParseHybridOverlayHostSubnet returns the parsed hybrid overlay hostsubnet if
+// the annotations included a valid one, or nil if they did not include one. If
+// one was included, but it is invalid, an error is returned.
+func ParseHybridOverlayHostSubnet(node *kapi.Node) (*net.IPNet, error) {
+	sub, ok := node.Annotations[types.HybridOverlayNodeSubnet]
+	if !ok {
+		return nil, nil
+	}
+	_, subnet, err := net.ParseCIDR(sub)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing node %s annotation %s value %q: %v",
+			node.Name, types.HybridOverlayNodeSubnet, sub, err)
+	}
+	return subnet, nil
+}
+
+// GetHybridOverlayPortName returns the name of the hybrid overlay switch port
+// for a given node
+func GetHybridOverlayPortName(nodeName string) string {
+	return "int-" + nodeName
+}
+
+// IsHybridOverlayNode returns true if the node has been labeled as a
+// node which does not participate in the ovn-kubernetes overlay network
+func IsHybridOverlayNode(node *kapi.Node) bool {
+	if config.Kubernetes.NoHostSubnetNodes != nil {
+		nodeSelector, _ := metav1.LabelSelectorAsSelector(config.Kubernetes.NoHostSubnetNodes)
+		return nodeSelector.Matches(labels.Set(node.Labels))
+	}
+	return false
+}
+
+func SameIPNet(a, b *net.IPNet) bool {
```

Comment:
cant you just do:
return a.String()==b.String()

Reply:
@trozet can't just do that because of nil and pointers, but addressed in a much better way than the existing function.